stable/lamp/templates/ingress.yaml

```diff
@@ -0,0 +1,126 @@
+{{- if .Values.ingress.enabled -}}
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: {{ template "fullname" . }}-app
+  annotations:
+    {{- if .Values.ingress.ssl }}
+    kubernetes.io/tls-acme: "true"
+    {{- end }}
+    ingress.kubernetes.io/proxy-body-size: "50m"
+    {{- if .Values.ingress.nginx }}
+    kubernetes.io/ingress.class: nginx
+    {{- end }}
+    {{- if .Values.ingress.htpasswd_string }}
+    ingress.kubernetes.io/auth-type: basic
+    ingress.kubernetes.io/auth-secret: {{ template "fullname" . }}
+    ingress.kubernetes.io/auth-realm: "Authenticate"
+    {{- end }}
+    {{- if .Values.ingress.www_subdomain }}
+    ingress.kubernetes.io/configuration-snippet: |
+      rewrite ^(.*)$ $scheme://www.{{ template "domain" . }}$1;
+    {{- end }}
+
+spec:
+  {{- if .Values.ingress.ssl }}
+  tls:
+  - secretName: {{ template "fullname" . }}-tls-app
+    hosts:
+    - {{ template "domain" . }}
+  {{- end }}
+  rules:
+  - host: {{ template "domain" . }}
+    http:
+      paths:
+      - path: /
+        backend:
+          serviceName: {{ template "fullname" . }}
+          servicePort: {{ .Values.service.http_port }}
+{{- if .Values.ingress.www_subdomain }}
+
+---
+
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: {{ template "fullname" . }}-www
+  annotations:
```

Comment:
Instead of wrapping various annotations in if blocks, I'd suggest you add the annotations to the configuration. This would also be more flexible.